autogpt_platform/backend/schema.prisma

```diff
@@ -272,11 +452,233 @@ model UserBlockCredit {
+
+  transactions UserBlockCredit[]
+
+  usdBalance Float @default(0.0)
+
+  subscription   UserSubscription? @relation(fields: [subscriptionId], references: [id])
+  subscriptionId String?
+}
+
+// This table tracks a user's subscription to autogpt
+model UserSubscription {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String
+  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  isSubscribed          Boolean   @default(false)
+  subscriptionStartDate DateTime?
+  subscriptionEndDate   DateTime?
+  isCancelling          Boolean   @default(false)
+
+  subscriptionPlanId String
+  subscriptionPlan   SubscriptionPlan @relation(fields: [subscriptionPlanId], references: [id])
+
+  // Stripe Integration Fields
+  stripeCustomerId         String?
+  stripeSubscriptionId     String?
+  stripePaymentMethodId    String?
+  stripeSubscriptionStatus String?
+  currentPeriodEnd         DateTime?
+
+  hasInitiatedChargeBackProcess Boolean          @default(false)
+  chargeBackProcessInitiatedAt  DateTime?
+  UserAccounting                UserAccounting[]
+
+  @@index([isSubscribed, isCancelling])
+}
+
+// This table tracks the different subscription plans available to a user
+model SubscriptionPlan {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  name        String
+  description String
+  planIconUrl String?
+
+  creditsPerMonth Int
+  usdPrice        Float
+
+  // Stripe Integration Field
+  stripePriceId String?
+
+  isDeleted        Boolean            @default(false)
+  UserSubscription UserSubscription[]
+}
+
+model StripeEvent {
+  id          String    @id @default(uuid())
+  eventId     String    @unique
+  eventType   String
+  createdAt   DateTime  @default(now())
+  processedAt DateTime?
+}
+
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+//////////////   Store TABLES    /////////////////////
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+
+model Profile {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String?
+  user   User?   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  isGroupProfile Boolean    @default(false)
+  groupId        String?
+  group          UserGroup? @relation(fields: [groupId], references: [id])
+
+  username    String @unique
+  description String
+
+  links String[]
+
+  avatarUrl String?
+
+  @@index([username])
+}
+
+model StoreListing {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  isDeleted  Boolean @default(false)
+  // Not needed but makes lookups faster
+  isApproved Boolean @default(false)
```

Comment:
If the information about approval is also somewhere else that creates multiple sources of truth and potentially needless bugs when that information becomes desynchronised (can it?). If that's the case is this optimization worth the risk? The same for `StoreListingVersion`.

Reply:
We don't want to over normalise the database here. The key is that there is only 1 source of truth, so if there ever is a disconnect between 2 locations, then we know which is correct.

In this case isApproved is based on the `StoreListingSubmission` how as isApproved is going to be checked for the majority of  queries on this table it makes sense to be able to index the table on it, rather than needing to do a join for all these queries.